screens/DoubleSearch.js

```diff
@@ -39,14 +39,24 @@ function DoubleSearch (props) {
         props.navigation.navigate("MapDirections");
     };
 
-    const selectDestinationPath = () => {
+    const selectDestinationName = () => {
         return props.navigation.getParam("destinationName", "null");
     };
 
-    //TODO: The new "From" location entered by the user will be stored in the asyncStorage with this key
-    AsyncStorage.setItem("fromLocation", from.toString());
-    // The new "To" destination can also be retrieved from the asyncStorage with this key
-    AsyncStorage.setItem("destination", to.toString());
+    let fromName = from.name;
+    let toName = to.name;
+
+    if (fromName !== undefined) {
```

Comment:
Why do we need to subscribe a state in AsyncStorage in this component? This component should max retrieve a value from AsyncStorage but not set any. This component interacts with the Directions component, but it can simply pass the required values as props. @csbduzi . To refactor.

Reply:
The reason for the call of the AsyncStorage as a setter is that the search bars of the DoubleSearch.js do not *only* interact with the *Direction* component. For the Direction component it can pass as a props because of the onPress function, however for the indoor maps that need to retrieve the values from the search bars do not have a direction link with DoubleSearch.js, meaning they will have to retrieve the values from an AsyncStorage

Reply:
Yeah I understand, but when we are going to merge both indoor and outdoor, I think we will pass those values to only one component. But very clear explanation, I understood why you added them. 💯